src/main_service/main_service.rs

```diff
@@ -1,48 +1,101 @@
-use std::{io::{Error, ErrorKind}, net::{TcpStream, ToSocketAddrs, UdpSocket}, os::unix::net::SocketAddr, sync::{atomic::{AtomicBool, Ordering}, mpsc::Sender, Arc, Mutex}, thread, time::Duration};
+use std::{io::{Error}, net::{ToSocketAddrs, UdpSocket}, os::unix::net::SocketAddr, sync::{atomic::{AtomicBool, Ordering}, mpsc::Sender, Arc, Mutex}, thread, time::Duration};
 use log::{info, warn};
-use sal_sync::services::{conf::conf_tree::ConfTree, entity::{
+use sal_sync::services::{conf::conf_tree::{self, ConfTree}, entity::{
         name::Name, object::Object, point::point::Point
         // services::{
         //     services::Services,
         //     service::service::Service,
         //     service::service_handles::ServiceHandles, 
         // }, 
 }, service::{service::Service, service_cycle::ServiceCycle, service_handles::ServiceHandles}, types::type_of::TypeOf};
-
-use crate::amplitude;
-
+use serde::de::value;
 use super::main_service_config::MainServiceConf;
-//
-//
-// impl ServiceConfig{
-//     //
-//     ///
-//     pub fn new(conf: &serde_yaml::Value) -> Self{
-//         let ampl = conf["ample"].as_i64().unwrap_or(0.0) as f64;
-//         let phi = conf["phi"].as_i64().unwrap_or(0.0) as f64;
-//         Self{
-//             ampl,
-//             phi,
-//         }
-//     }
-// }
-//
-//
+use crate::Buffer;
+///
+/// Struct `UpdHeader`
+/// - `syn` - message starts with
+/// - `addr` - address of the input channel
+/// - `type` - type of values in the array in data field in struct `UpdMessage`
+/// - `count` - length of the array in the data field in struct `UpdMessage`
 pub struct UdpHeader {
     pub syn: u8,
     pub addr: u8,
     pub r#type: u8,
     pub count: u8,
-    pub data: [u8; 1024],
 }
+//
+//
+impl UdpHeader{
+    ///
+    /// Creates a header for udp
+    pub fn new(syn: u8, addr: u8, r#type: u8, count:  u8) -> Self{
+        Self{
+            syn,
+            addr,
+            r#type,
+            count,
+        }
+    }
+    ///
+    /// Convert fields of UdpHeader to Vector
+    pub fn to_bytes(&self) -> Vec<u8>{
+        let mut header_bytes = Vec::with_capacity(4);
+        header_bytes.push(self.syn);
+        header_bytes.push(self.addr);
+        header_bytes.push(self.r#type);
+        header_bytes.push(self.count);
+        header_bytes
+    }
+}
+///
+/// Struct `UpdMessage`
+/// - `header` - contains the UPD header information
+/// - `data` - array of values
+pub struct UpdMessage{
```

Comment:
Move to the `udp_message.rs` file
- develop in the Issie-PR pair
- may bee unit test ?